binary_search_tree/tree.py

```diff
@@ -9,40 +9,121 @@ def __init__(self, key, val = None):
         self.right = None
         
 
-
 class Tree:
     def __init__(self):
         self.root = None
 
+    def addHelper(self, current_root, new_node):
+        if current_root is None:
+            return new_node
+        
+        if new_node.key < current_root.key:
+            if current_root.left is None:
+                # current_root.left = self.addHelper(current_root.left, new_node)
+                current_root.left = new_node
+            else:
+                self.addHelper(current_root.left, new_node)
+        else:
+            if current_root.right is None:
+                # current_root.right = self.addHelper(current_root.right, new_node)
+                current_root.right = new_node
+            else:
+                self.addHelper(current_root.right, new_node)
+
     # Time Complexity: 
     # Space Complexity: 
     def add(self, key, value = None):
```

Comment:
👀 Good job implementing the function. But what would the time and space complexity be? 🤔